### PR TITLE
feat: ROOT LinkDef `vector<vector<int>>` and co in Examples

### PR DESCRIPTION
**Pull request by andiwand**
Since this popped up a couple of times I would like to add `vector<vector<int>>` and co to ROOT IO

### CODE REVIEW

**Comment by andiwand** [comment]
lets see if `uproot` and `TBrowser` can eat this

**Comment by paulgessinger** [comment]
TBrowser should handle this alright if you have the dictionary on your ROOT path, uproot I don't think will be able to read these.

**Comment by andiwand** [comment]
seems to work with `TBrowser` and `uproot`

image

```
In [9]: ak.to_dataframe(f["vertices"].arrays(library="ak"), how="outer")
Out[9]: 
                            event_id         vertex_id  ...  vertex_secondary  generation
entry subentry subsubentry                              ...                              
0     0        0                   0  4503599627370496  ...                 0           0
               1                   0  4503599627370496  ...                 0           0
               2                   0  4503599627370496  ...                 0           0
               3                   0  4503599627370496  ...                 0           0
1     0        0                   1  4503599627370496  ...                 0           0
...                              ...               ...  ...               ...         ...
98    0        3                  98  4503599627370496  ...                 0           0
99    0        0                  99  4503599627370496  ...                 0           0
               1                  99  4503599627370496  ...                 0           0
               2                  99  4503599627370496  ...                 0           0
               3                  99  4503599627370496  ...                 0           0

[400 rows x 11 columns]
```

**Comment by andiwand** [comment]
`rootcling` is crashing on `linux-nodeps` https://github.com/acts-project/acts/actions/runs/8093259470/job/22115536288?pr=3000#step:14:177

**Comment by paulgessinger** [comment]
@andiwand It's possible they ship generic code for this. If there are no warnings/errors printed, I'm happy to add this.

**Comment by andiwand** [comment]
@AJPfleger is there a way to silence this error? https://github.com/acts-project/acts/actions/runs/8093259472/job/22115538516?pr=3000#step:4:546

File: CI/check_unused_files.py [hunk 1]
```diff
@@ -59,6 +59,8 @@ def main():
         # TODO Mention these files somewhere?
         "generate_particle_data_table.py",
         "lazy_autodoc.py",
+        # ROOT LinkDef.h files are referenced by CMakeLists.txt and incorrectly detected as unused
+        "LinkDef.h",
```
Comment (andiwand) [inline CI/check_unused_files.py:63]:
```suggestion
```

File: Examples/Io/Root/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,16 @@
+add_library(
+  ActsExamplesIoRootLinkDef OBJECT)
+target_include_directories(
+  ActsExamplesIoRootLinkDef
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+target_link_libraries(
+  ActsExamplesIoRootLinkDef
+  PRIVATE ROOT::RIO)
+set_target_properties(
+  ActsExamplesIoRootLinkDef PROPERTIES CXX_CLANG_TIDY "")
```
Comment (andiwand) [inline Examples/Io/Root/CMakeLists.txt:10]:
```suggestion
set_target_properties(
  ActsExamplesIoRootLinkDef PROPERTIES C_CLANG_TIDY "" CXX_CLANG_TIDY "")
```

File: Examples/Io/Root/CMakeLists.txt [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
+target_include_directories(
+  ActsExamplesIoRootLinkDef
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+target_link_libraries(
+  ActsExamplesIoRootLinkDef
+  PRIVATE ROOT::RIO)
+set_target_properties(
+  ActsExamplesIoRootLinkDef PROPERTIES C_CLANG_TIDY "" CXX_CLANG_TIDY "")
+ROOT_GENERATE_DICTIONARY(
+  ActsDicts MODULE ActsExamplesIoRootLinkDef LINKDEF LinkDef.hpp)
```
Comment (andiwand) [inline Examples/Io/Root/CMakeLists.txt:12]:
```suggestion
ROOT_GENERATE_DICTIONARY(
  ActsDicts MODULE ActsExamplesIoRootLinkDef LINKDEF LinkDef.hpp)
set_target_properties(
  ActsExamplesIoRootLinkDef PROPERTIES CXX_CLANG_TIDY "")
```

File: Examples/Io/Root/CMakeLists.txt [hunk 4]
```diff
@@ -18,6 +31,7 @@ add_library(
   src/RootTrackSummaryWriter.cpp
   src/RootBFieldWriter.cpp
   src/RootAthenaNTupleReader.cpp
+  $<TARGET_OBJECTS:ActsExamplesIoRootLinkDef
```
Comment (andiwand) [inline Examples/Io/Root/CMakeLists.txt:34]:
```suggestion
  $<TARGET_OBJECTS:ActsExamplesIoRootLinkDef>
```

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [fd0e8453e7474e0fcfdeb60ddb4408d6a6b722f5]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/36718007) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7002957)!

**Pull request merged**